pkg/transaction/transaction_test.go

```diff
@@ -90,13 +89,15 @@ func TestTransactionSend(t *testing.T) {
 	t.Run("send", func(t *testing.T) {
 		t.Parallel()
 
-		signedTx := types.NewTx(&types.LegacyTx{
-			Nonce:    nonce,
-			To:       &recipient,
-			Value:    value,
-			Gas:      estimatedGasLimit,
-			GasPrice: suggestedGasPrice,
-			Data:     txData,
+		signedTx := types.NewTx(&types.DynamicFeeTx{
+			ChainID:   chainID,
+			Nonce:     nonce,
+			To:        &recipient,
+			Value:     value,
+			Gas:       estimatedGasLimit,
+			GasTipCap: suggestedGasPrice,
+			GasFeeCap: suggestedGasPrice,
```

Comment:
For the test, do we want to separate the GasTipCap and GasFeeCap to 2 different values to ensure that they are properly carried through and not mixed?